intrastat_product/models/intrastat_product_declaration.py

```diff
@@ -294,70 +285,42 @@ def _get_partner_country(self, inv_line, notedict, eu_countries):
                     % (inv.name, country.name)
                 ]
                 self._format_line_note(inv_line, notedict, line_notes)
-        if country and country.code == "GB" and self.year >= "2021":
-            vat = inv.commercial_partner_id.vat
-            if not vat:
-                line_notes = [
-                    _(
-                        "On invoice '%s', the source/destination country "
-                        "is United-Kingdom and the fiscal position is '%s'. "
-                        "Make sure that the fiscal position is right. If "
-                        "the origin/destination is Northern Ireland, please "
-                        "set the VAT number of the partner '%s' in Odoo with "
-                        "its new VAT number starting with 'XI' following Brexit."
-                    )
-                    % (
-                        inv.name,
-                        inv.fiscal_position_id.display_name,
-                        inv.commercial_partner_id.display_name,
-                    )
-                ]
-                self._format_line_note(inv_line, notedict, line_notes)
-            elif not vat.startswith("XI"):
-                line_notes = [
-                    _(
-                        "On invoice '%s', the source/destination country "
-                        "is United-Kingdom, the fiscal position is '%s' and "
-                        "the partner's VAT number is '%s'. "
-                        "Make sure that the fiscal position is right. If "
-                        "the origin/destination is Northern Ireland, please "
-                        "update the VAT number of the partner '%s' in Odoo with "
-                        "its new VAT number starting with 'XI' following Brexit."
-                    )
-                    % (
-                        inv.name,
-                        inv.fiscal_position_id.display_name,
-                        vat,
-                        inv.commercial_partner_id.display_name,
-                    )
-                ]
-                self._format_line_note(inv_line, notedict, line_notes)
```

Comment:
Can we maintain these warnings with this new approach?

Reply:
Yes, we could keep the second warning and restrict it for B2B (vat_required=True on fiscal position). We should drop the first warning (or make it generic for B2B when vat_required=True on the fiscal position.